### ISSUE TITLE
Error when opening the Treesitter inspector and the first line is removed

### ISSUE THREAD

**Issue by MariaSolOs** [neovim/neovim#24892]
### Problem

I'm reporting this super early because the change isn't even on nightly, I just found it when working on #24890.

Using the tippy-top of `master`, opening the tree inspector causes the error below and the first line in the buffer is removed.

```
Error detected while processing CursorMoved Autocommands for "<buffer=53>":
Error executing lua callback: runtime/lua/vim/treesitter/dev.lua:372: Invalid 'end_col': out of range
stack traceback:
	[C]: in function 'nvim_buf_set_extmark'
	runtime/lua/vim/treesitter/dev.lua:372: in function <runtime/lua/vim/treesitter/dev.lua:368>
```

Undoing #24824 (cc @bfredl) makes the error go away and the inspector to behave as expected.

### Steps to reproduce

1. Build the `master` branch of this repo.
2. Open any file.
3. Run `:InspectTree`.

### Expected behavior

For no errors to occur and for the tree inspector to not modify the inspected buffer.

### Neovim version (nvim -v)

N/A

### Vim (not Nvim) behaves the same?

No

### Operating system/version

Mac Ventura 13.5

### Terminal name/version

kitty 0.28.1

### $TERM environment variable

xterm-kitty

### Installation

build from repo

**Comment by MariaSolOs** [comment]
Ohhh my bad @bfredl, I'm checking out if https://github.com/neovim/neovim/pull/24889 fixes this.

**Comment by aarondill** [comment]
I'm experiencing what I think is this same bug, though the [noice.nvim](https://github.com/folke/noice.nvim) plugin. Whenever I save a file, the first line is removed. It doesn't happen if I disable noice.nvim (though a lot of things change behavior if I do, so it may still be another plugin). This doesn't happen on v0.9.1, only on the nightly.

**Comment by aarondill** [comment]
I've done a bisect and confirmed that 008154954791001efcc46c28146e21403f3a698b is the first bad commit

**Comment by bfredl** [comment]
@MariaSolOs would you mind test https://github.com/neovim/neovim/pull/24889 again?

**Comment by wookayin** [comment]
Still happens after #24889 (on 63802a1); The first line being removed. This is unawarely destructing many files that are not VCS-tracked 😂

**Comment by MariaSolOs** [comment]
@bfredl #24889 is still not working with the inspector for me :/

**Comment by wookayin** [comment]
Reported in #24911 and fixed by #24913

**Comment by MariaSolOs** [comment]
@wookayin I confirm that this issue is now fixed on `master`.
@bfredl thank you for the fix!
